mozilla_aws_cli/role_picker.py

```diff
@@ -11,26 +11,64 @@
 
 logger = logging.getLogger(__name__)
 
-PROMPT_BASH_CODE = r'''function maws_profile {
-    if [ -n "${MAWS_PROMPT}" ]; then
-        if [ -n "${AWS_SESSION_EXPIRATION}" ] && [ $(date +%s) -gt ${AWS_SESSION_EXPIRATION} ]; then
-            echo " (maws keys expired)"
+PROMPT_BASH_CODE = r'''
+function maws_profile {
+    if [[ -n $MAWS_PROMPT ]]; then
```

Comment:
> I'd used `[` instead of `[[` to be most broadly compatible. [`[`](https://pubs.opengroup.org/onlinepubs/9699919799/utilities/test.html) is a POSIX standard whereas [`[[`](https://www.gnu.org/software/bash/manual/html_node/Conditional-Constructs.html#index-_005b_005b) is a bash extension that comes from KSH.
> 
> What would you think about retaining the POSIX standard `[` across this code?

The original code for PROMPT_BASH_CODE already took for granted that the shell was bash or bash-compatible (the use of `$(date +%s)`, specifically), and would not only fail with syntax errors but also fail to modify the user's prompt (the assumption that `PS1` is the prompt), each time they eval the output of maws; in a brief chat with someone else, they just don't eval maws output and grab the environment variables instead. I don't expect this patch makes that scenario any worse for them, and I'm not taking "support non-bash-compatible prompts" into the scope of the pull request.

Reply:
> Also it's [always good to quote shell variables](https://github.com/git/git/blob/d486ca60a51c9cb1fe068803c3f540724e95e83a/contrib/completion/git-prompt.sh#L127) to avoid unexpected behavior when the value in the variable contains spaces or other oddities.

When working with bash-specific conditionals inside [[ ... ]], overquoting not only is unnecessary but can break functionality of regular expressions and string-prefix/suffix comparisons. The specified bash code passes all shellcheck reviews and the only warning is an [SC2016](https://github.com/koalaman/shellcheck/wiki/SC2016) false positive, as we're specifically avoiding string interpolation in a case where virtually everyone else is expecting to interpolate (because we're deferring interpolation until prompt eval time).

I would be more wary if this were not `PROMPT_BASH_CODE` and were instead meant for other shells, but as above, this has always taken for granted bash and expanding or repairing that constraint is out of scope here.